src/timezones/time.jl

```diff
@@ -0,0 +1,76 @@
+# Convenience type for working with HH:MM:SS.
+immutable Time <: TimePeriod
+    seconds::Int
+end
+const ZERO = Time(0)
+
+function Time(hour::Int, minute::Int, second::Int)
+    Time(hour * 3600 + minute * 60 + second)
+end
+
+function Time(s::String)
+    # "-" represents 0:00 for some DST rules
+    s == "-" && return ZERO
+    parsed = map(n -> parse(Int, n), split(s, ':'))
+
+    # Only can handle up to hour, minute, second.
+    length(parsed) > 3 && error("Invalid Time string")
+    any(parsed[2:end] .< 0) && error("Invalid Time string")
+
+    # Handle variations where minutes and seconds may be excluded.
+    values = [0,0,0]
+    values[1:length(parsed)] = parsed
+
+    if values[1] < 0
+        for i in 2:length(values)
+            values[i] = -values[i]
+        end
+    end
+
+    return Time(values...)
+end
+
+# TimePeriod methods
+Base.Dates.value(t::Time) = t.seconds
+Base.Dates.toms(t::Time) = t.seconds * 1000
+
+toseconds(t::Time) = t.seconds
+hour(t::Time) = div(toseconds(t), 3600)
+minute(t::Time) = rem(div(toseconds(t), 60), 60)
+second(t::Time) = rem(toseconds(t), 60)
+
+function hourminutesecond(t::Time)
+    h, r = divrem(toseconds(t), 3600)
+    m, s = divrem(r, 60)
+    return h, m, s
+end
+
+Base.convert(::Type{Second}, t::Time) = Second(toseconds(t))
+Base.convert(::Type{Millisecond}, t::Time) = Millisecond(toseconds(t) * 1000)
+Base.promote_rule{P<:Union{Week,Day,Hour,Minute,Second}}(::Type{P}, ::Type{Time}) = Second
+Base.promote_rule(::Type{Millisecond}, ::Type{Time}) = Millisecond
+
+# Should be defined in Base.Dates
+Base.isless(x::Period, y::Period) = isless(promote(x,y)...)
```

Comment:
Yeah, I think there was trepidation about defining `promote_rule` for Period types because of `Year`/`Month`. I always figured we could define it later with better definition/bounds on what was legal and what wasn't.